core-bundle/src/Resources/contao/elements/ContentMedia.php

```diff
@@ -130,22 +130,27 @@ protected function compile()
 		// Convert the language to a locale (see #5678)
 		$strLanguage = LocaleUtil::formatAsLocale($objPage->language);
 
+		// Determine caption
+		$strCaption = $this->playerCaption;
+
 		// Pass File objects to the template
 		while ($objFiles->next())
 		{
-			$arrMeta = StringUtil::deserialize($objFiles->meta);
+			$objMeta = $objFiles->current()->getMetadata($strLanguage);
```

Comment:
I'd prefer if we would rewrite L137 to a `foreach($objFiles as $fileModel)` instead of using `current()`. This is IMHO less likely to introduce subtle bugs.

Reply:
Agreed, but I wanted to keep the PR minimal. This should be done in a separate PR and replace the `while` with `foreach` in all other places as well.

Reply:
I'm not sure it's worth refactoring everywhere in the legacy code. But it's probably easy enough at the places where new changes are introduced? But up to you.

For reference in https://github.com/contao/contao/pull/2788/files we fixed the bug I introduced earlier on that could have been avoided if I had done the minimal refactoring in the first place. :smile:

Reply:
I'm with @m-vo on this.

Reply:
Done :)